core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -1135,6 +1136,71 @@ object TestUtils extends Logging {
   }
 
   /**
+   * Verifies that this ACL is the secure one. The
+   * values are based on the constants used in the
+   * ZooKeeper code base.
+   */
+  def isAclSecure(acl: ACL, sensitive: Boolean): Boolean = {
+    info(s"ACL $acl")
+    acl.getPerms match {
+      case 1 => !sensitive && acl.getId.getScheme.equals("world")
+      case 31 => acl.getId.getScheme.equals("sasl")
+      case _ => false
+    }
+  }
+
+  /**
+   * Verifies that the ACL corresponds to the unsecure one.
+   */
+  def isAclUnsecure(acl: ACL): Boolean = {
+    info(s"ACL $acl")
+    acl.getPerms match {
+      case 31 => acl.getId.getScheme.equals("world")
+      case _ => false
+    }
+  }
+
+  private def secureZkPaths(zkUtils: ZkUtils) = {
+    def subPaths(path: String) = {
+      if (zkUtils.pathExists(path))
+        zkUtils.getChildren(path).map(c => path + "/" + c)
```

Comment:
Sorry, my question was whether we should check all the paths instead of stopping at the direct children. In other words, call `subPaths` for each child.